yarn-project/end-to-end/src/e2e_auth_contract.test.ts

```diff
@@ -0,0 +1,125 @@
+import { AccountWallet, AztecAddress, ContractFunctionInteraction, Fr, PXE } from '@aztec/aztec.js';
+import { AuthContract } from '@aztec/noir-contracts.js';
+
+import { jest } from '@jest/globals';
+
+import { publicDeployAccounts, setup } from './fixtures/utils.js';
+
+describe('e2e_auth_contract', () => {
+  const TIMEOUT = 120_000;
+  jest.setTimeout(TIMEOUT);
+
+  let teardown: () => Promise<void>;
+
+  let admin: AccountWallet;
+  let authorized: AccountWallet;
+  let other: AccountWallet;
+
+  let pxe: PXE;
+
+  let contract: AuthContract;
+
+  const VALUE = 3;
+  const DELAY = 5;
+
+  beforeAll(async () => {
+    ({
+      teardown,
+      wallets: [admin, authorized, other],
+      pxe,
+    } = await setup(3));
+
+    await publicDeployAccounts(admin, [admin.getAddress(), authorized.getAddress(), other.getAddress()]);
+
+    const deployTx = AuthContract.deploy(admin, admin.getAddress()).send({});
+    const receipt = await deployTx.wait();
+    contract = receipt.contract;
+  });
+
+  afterAll(() => teardown());
+
+  async function mineBlock() {
+    await contract.methods.get_authorized().send().wait();
+  }
+
+  async function mineBlocks(amount: number) {
+    for (let i = 0; i < amount; ++i) {
+      await mineBlock();
+    }
+  }
+
+  async function assertLoggedAddress(interaction: ContractFunctionInteraction, address: AztecAddress) {
+    const logs = await pxe.getUnencryptedLogs({ txHash: (await interaction.send().wait()).txHash });
```

Comment:
We have a [getUnencryptedLogs method](https://github.com/AztecProtocol/aztec-packages/blob/5af2b95a477966d4068741a60a25f8da433abecf/yarn-project/aztec.js/src/contract/sent_tx.ts#L100) on `SentTx` and it would make this a bit nicer.

It's just a detail but since devs use these e2e tests as reference it would be nice to demonstrate it here.

Reply:
Hm the problem is that in some tests I want to keep the interaction and not send the tx (since e.g. i will simulate it). I'll review this API later, thanks for pointing this out!